src/DataModel/IData.ts

```diff
@@ -0,0 +1,6 @@
+import {IMetadata} from "./IMetadata";
+
+export interface IData extends Object
```

Comment:
I agree, calling this `IResource` would seem better. I don't think `IResource` should be split in data and metadata (it's hard to draw that line anyway). Having methods to easily access subjects of interest should do the job (`getProperty`, `getOperation`, ...)